components/ClickAwayListener/ClickAwayListener.tsx

```diff
@@ -0,0 +1,20 @@
+import React, { FC, ReactElement, RefObject, useRef } from 'react';
+import useOnClickAway from 'Hooks/useOnClickAway';
+
+type Props = {
+  onClickAway: () => void;
+};
+
+const ClickAwayListener: FC<Props> = ({ children, onClickAway }) => {
+  const ref = useRef<HTMLElement>(null);
+  useOnClickAway(ref, () => onClickAway());
+
+  if (React.Children.count(children) == 1) {
+    const copyWithRef = React.cloneElement(children as ReactElement, { ref: ref });
+    return <React.Fragment>{copyWithRef}</React.Fragment>;
+  }
+
+  return <div ref={ref as RefObject<HTMLDivElement>}>{children}</div>;
```

Comment:
In general, if we think we have to use type assertion in TS (tell that "this value here is for 100% good, please be silent mr. compiler" 😅), for 99% our code can be improved or even redesigned. 

In general, good goal while writing TS is to avoid interpolation 99% of the time (there are exceptions like [narrowing types while using DOM operation methods like `document.querySelector()`](https://www.typescriptlang.org/docs/handbook/2/everyday-types.html#type-assertions=)).